calicoctl/calico_ctl/pool.py

```diff
@@ -118,18 +118,32 @@ def ip_pool_add(cidrs, version, ipip, masquerade):
         print "IP in IP not supported for IPv6 pools"
         sys.exit(1)
 
-    # TODO Reject any cidrs that overlap with existing cidrs in the pool
-    pools = []
-    for cidr in cidrs:
+    current_pools = client.get_ip_pools(version)
+    current_pool_nets = [IPNetwork(pool.cidr) for pool in current_pools]
+
+    new_pools = []
+
+    for new_cidr in cidrs:
+        # Check if new pool overlaps with any existing pool
+        new_cidr_net = IPNetwork(new_cidr)
+        for cur_pool_net in current_pool_nets:
+
+            if cur_pool_net in new_cidr_net or new_cidr_net in cur_pool_net:
```

Comment:
probably need to check the supplied pools don't overlap with each other too.

and typo below:  exisitng